protoarray/src/main/java/tech/pegasys/teku/protoarray/ProtoArray.java

```diff
@@ -51,25 +51,39 @@
    * <p>The {@link #removeBlockRoot(Bytes32)} method removes blocks from this map but does not
    * change the nodes list to avoid having to adjust all indices.
    */
-  private final Map<Bytes32, Integer> indices;
+  private final ProtoArrayIndices protoArrayIndices = new ProtoArrayIndices();
```

Comment:
nit: I'd be very tempted to still call this just `indices`.  The protoarray is pretty clearly implied since we're in a class called `ProtoArray`.